pkg/stores/sqlproxy/proxy_store.go

```diff
@@ -676,53 +533,43 @@ func (s *Store) ListByPartitions(apiOp *types.APIRequest, schema *types.APISchem
 		return nil, 0, "", err
 	}
 
-	return list.Items, total, continueToken, nil
+	return list, total, continueToken, nil
 }
 
 // WatchByPartitions returns a channel of events for a list or resource belonging to any of the specified partitions
-func (s *Store) WatchByPartitions(apiOp *types.APIRequest, schema *types.APISchema, wr types.WatchRequest, partitions []partition.Partition) (chan watch.Event, error) {
-	ctx, cancel := context.WithCancel(apiOp.Context())
-	apiOp = apiOp.Clone().WithContext(ctx)
-
-	eg := errgroup.Group{}
-
-	result := make(chan watch.Event)
+func (s *Store) WatchByPartitions(apiOp *types.APIRequest, schema *types.APISchema, wr types.WatchRequest, partitions []partition.Partition) (chan struct{}, error) {
+	ctx := apiOp.Context()
 
-	for _, partition := range partitions {
-		p := partition
-		eg.Go(func() error {
-			defer cancel()
-			c, err := s.watchByPartition(p, apiOp, schema, wr)
-
-			if err != nil {
-				return err
-			}
-			for i := range c {
-				result <- i
-			}
-			return nil
-		})
+	revision := 0
+	if wr.Revision != "" {
+		parsedRevision, err := strconv.ParseInt(wr.Revision, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("invalid revision %q: %w", wr.Revision, err)
+		}
+		revision = int(parsedRevision)
 	}
 
-	go func() {
-		defer close(result)
-		<-ctx.Done()
-		eg.Wait()
-		cancel()
-	}()
+	// XXX: Why was this needed at all??
+	apiOp = apiOp.Clone().WithContext(ctx)
 
-	return result, nil
-}
+	// warnings from inside the informer are discarded
+	buffer := WarningBuffer{}
+	client, err := s.clientGetter.TableAdminClient(apiOp, schema, "", &buffer)
+	if err != nil {
+		return nil, err
+	}
 
-// watchByPartition returns a channel of events for a list or resource belonging to a specified partition
-func (s *Store) watchByPartition(partition partition.Partition, apiOp *types.APIRequest, schema *types.APISchema, wr types.WatchRequest) (chan watch.Event, error) {
-	if partition.Passthrough {
-		return s.Watch(apiOp, schema, wr)
+	inf, err := s.CacheFor(&tablelistconvert.Client{ResourceInterface: client}, schema)
+	if err != nil {
+		return nil, err
 	}
 
-	apiOp.Namespace = partition.Namespace
-	if partition.All {
-		return s.Watch(apiOp, schema, wr)
+	debounceListener := newDebounceListener(5 * time.Second)
```

Comment:
Of course we will have to make sure the listener time is configurable - at least via configuration object or environment variable, in future possibly on a request-by-request basis.